docs/DeveloperGenerateTest.md

```diff
@@ -71,19 +104,198 @@ var _ = framework.ChaosSuiteDescribe("Chaos tests", Label("Chaos"), func() {
+ 		* each string intended to be a label MUST be prefixed with `@`
+    * the set of labels MUST be a comma separated list
+    * they MUST be assigned AFTER the description text
+ * When using the `DescribeTable` key, the proceeding nested lines MUST have the `Entry` or `By` key or Ginkgo will not render the template properly
+
+For the time being we don't support any of Ginkgo's Setup/Teardown nodes. We could technically graph it together from the text outline but it won't render with our base template. The important thing is to expressively model the behavior to test. Test developers will be able to insert Setup/Teardown nodes where they see fit when the spec has been rendered. 
 
- ## Generating Ginkgo Test Suite File
+
+## Prerequisite
+
+Before generating anything make sure you have Ginkgo in place. To install Ginkgo:
+
+`go install -mod=mod github.com/onsi/ginkgo/v2/ginkgo`
+
+ 
+ ## Usage
+
+ ### Printing a text outline of an existing ginkgo spec file
+ This will generate the outline and output to your terminal. If you would like to save the outline to a file refer to the next section.
+
+`./mage PrintOutlineOfGinkgoSpec tests/<subdirectory>/<test-file>.go`
+
+```bash
+$ ./mage PrintOutlineOfGinkgoSpec tests/books/books.go
+tests/books/books.goI0622 22:11:30.391491   19273 testcasemapper.go:26] Mapping outline from a Ginkgo test file, tests/books/books.go
+I0622 22:11:30.395240   19273 testcasemapper.go:40] Printing outline:
+
+BookSuiteDescribe: Book service E2E tests
+  Describe: Categorizing book length  @book
+    When: the book has more than 300 pages  @slow
+      It: Should be a novel
+    When: the book has fewer than 300 pages  @fast
+      It: should be a short story
+
+  Describe: Creating bookmarks in a book  @book, @bookmark, @parallel
+    It: Has no bookmarks by default
+    It: Can add bookmarks
+
+  DescribeTable: Reading invalid books always errors is table
+    Entry: Empty book
+    Entry: Only title
+    Entry: Only author
+    Entry: Missing pages
+
+```
+### Generating the text outline file from an existing ginkgo spec file
+This will generate the outline and output to a text file to the desired file location
+
+`./mage GenerateTextOutlineFromGinkgoSpec tests/<subdirectory>/<test-file>.go <dest>/<sub-path>/<file>`
+
+```bash
+$ ./mage GenerateTextOutlineFromGinkgoSpec tests/books/books.go /tmp/outlines/books.outline
+I0622 22:19:41.857966   20923 testcasemapper.go:26] Mapping outline from a Ginkgo test file, tests/books/books.go
+I0622 22:19:41.865725   20923 testcasemapper.go:70] Mapping outline to a text file, /tmp/outlines/books.outline
+I0622 22:19:41.865837   20923 textspec.go:67] successfully written to /tmp/outlines/books.outline
+
+```
+ 
+ ### Generating the ginkgo spec file from an existing text outline file
+ This will generate the Ginkgo spec in a subdirectory within our tests directory
+
+ `./mage GenerateGinkoSpecFromTextOutline <path>/<to>/<outline> <subpath-under-tests>/<filename>.go`
```

Comment:
I'm hitting this error when running
```
❯ ./mage GenerateGinkoSpecFromTextOutline outline tests/test/test.go
I0704 11:10:10.625582   81559 magefile.go:592] Mapping outline from a text file, outline
I0704 11:10:10.625969   81559 magefile.go:599] Mapping outline to a Ginkgo spec file, tests/test/test.go
I0704 11:10:10.626197   81559 ginkgosspec.go:178] /Users/psturc/work/redhat-appstudio/e2e-tests/tests
I0704 11:10:10.627840   81559 ginkgosspec.go:134] Creating new test package directory and spec file /Users/psturc/work/redhat-appstudio/e2e-tests/tests/test/test.go.
ginkgo generate failed
  Failed to read custom template file:
  open ../../templates/test_output_spec.tmpl: no such file or directory
```

Reply:
@psturc check now. I forgot to include the updated template file when I created the new branch. Let me know if works now.

Reply:
:sigh: check now. I forgot to push up the new file.

Reply:
works ok now 👍